org.contextmapper.dsl/src/org/contextmapper/dsl/validation/ApplicationCoordinationSemanticsValidator.java

```diff
@@ -0,0 +1,107 @@
+/*
+ * Copyright 2019 The Context Mapper Project Team
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.contextmapper.dsl.validation;
+
+import static org.contextmapper.dsl.validation.ValidationMessages.COORDINATION_STEP_CONTEXT_NOT_REACHABLE;
+import static org.contextmapper.dsl.validation.ValidationMessages.COORDINATION_STEP_SERVICE_NOT_ON_STEP_CONTEXT_APPLICATION;
+import static org.contextmapper.dsl.validation.ValidationMessages.COORDINATION_STEP_OPERATION_NOT_ON_STEP_SERVICE;
+import static org.contextmapper.dsl.validation.ValidationMessages.COORDINATION_STEP_OPERATION_IS_AMBIGUOUS;
+import static org.contextmapper.dsl.validation.ValidationMessages.VISUALIZE_COORDINATION_WITH_SKETCH_MINER;
+
+import java.util.List;
+
+import org.contextmapper.dsl.cml.CMLModelObjectsResolvingHelper;
+import org.contextmapper.dsl.contextMappingDSL.BoundedContext;
+import org.contextmapper.dsl.contextMappingDSL.ContextMappingDSLPackage;
+import org.contextmapper.dsl.contextMappingDSL.ContextMappingModel;
+import org.contextmapper.dsl.contextMappingDSL.Coordination;
+import org.contextmapper.dsl.contextMappingDSL.CoordinationStep;
+import org.contextmapper.tactic.dsl.tacticdsl.Service;
+import org.contextmapper.tactic.dsl.tacticdsl.ServiceOperation;
+import org.eclipse.xtext.EcoreUtil2;
+import org.eclipse.xtext.validation.AbstractDeclarativeValidator;
+import org.eclipse.xtext.validation.Check;
+import org.eclipse.xtext.validation.EValidatorRegistrar;
+
+public class ApplicationCoordinationSemanticsValidator extends AbstractDeclarativeValidator {
+	
+	public static final String SKETCH_MINER_INFO_ID = "open-coordination-in-sketch-miner";
+
+	@Override
+	public void register(EValidatorRegistrar registrar) {
+		// not needed for classes used as ComposedCheck
+	}
+	
+	@Check
+	public void stepComponentsMustBeCorrectlyReferenced(final CoordinationStep coordinationStep) {
```

Comment:
Thats a very long method with multiple checks. Could you maybe split it into multiple methods? That blocks that are marked with comments could be separate methods I guess.